charts/seed-controlplane/charts/etcd/templates/etcd.yaml

```diff
@@ -0,0 +1,69 @@
+apiVersion: druid.gardener.cloud/v1alpha1
+kind: Etcd
+metadata:
+  name: etcd-{{ .Values.role }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    garden.sapcloud.io/role: controlplane
+    role: {{ .Values.role }}
+{{- if .Values.labels }}
+{{ toYaml .Values.labels | indent 4 }}
+{{- end }}
+spec:
+{{- if .Values.podAnnotations }}
+  annotations:
+{{ toYaml .Values.podAnnotations | indent 4 }}
+{{- end }}
+  labels:
+    garden.sapcloud.io/role: controlplane
+    app: etcd-statefulset
+    role: {{ .Values.role }}
+    networking.gardener.cloud/to-dns: allowed
+    networking.gardener.cloud/to-public-networks: allowed
+    networking.gardener.cloud/to-private-networks: allowed
+  etcd:
+    tls:
+      serverTLSSecretRef:
+        name: {{ .Values.tlsServerSecretName }}
+        namespace: {{ .Release.Namespace }}
+      clientTLSSecretRef:
+        name: {{ .Values.tlsClientSecretName }}
+        namespace: {{ .Release.Namespace }}
+    metrics: {{ .Values.metrics }}
+    version: {{ .Values.etcdVersion }}
+    defragmentationSchedule: '0 */24 * * *'
+{{- if .Values.etcdResources }}
+    resources:
+{{ toYaml .Values.etcdResources | indent 6 }}
+{{- end }}
+    clientPort: {{ .Values.servicePorts.client }}
+    serverPort: {{ .Values.servicePorts.server }}
+  backup:
+    version: {{ .Values.brVersion }}
+    port: 8080
+    fullSnapshotSchedule: "0 */24 * * *"
+{{- if .Values.brResources }}
+    resources:
+{{ toYaml .Values.brResources | indent 6 }}
+{{- end }}
+    garbageCollectionPolicy: Exponential
+    Quota: 8Gi
+    garbageCollectionPeriod: 43200s
+    deltaSnapshotPeriod: 300s
```

Comment:
Shouldn't these two also be conditional?

`{{- if .Values.backupEnabled }}`

Reply:
`backupEnabled` is set if Seed.Spec.Backup is not `nil`. That has info about the backup bucket. `garbageCollectionPeriod` and `deltaSnapshotPeriod` are needed for local snapshots also.

Reply:
I don't understand. Why are delta snapshots (and `deltaSnapshotMemoryLimit` setting) needed if backup is not enabled? (I was not talking about `garbageCollectionPeriod`).

Reply:
If the provider is "" then the sidecar will not use `deltaSnapshotMemoryLimit` and `deltaSnapshotMemoryLimit`. I have kept the provider value `local`. Instead I'll make it "". In this case, these values not be acted upon. If it has to be totally omitted from the crd spec then Ill have to do a check on the `role` value as well.

Reply:
OK, fine, I'm just saying that from an end-user's PoV it's quite confusing that you have to write `deltaSnapshotPeriod: ""` and `deltaSnapshotMemoryLimit: ""` if you are not using a backup. It's cleaner to just omit the fields.

Reply:
Done